src/client/rpc_api.rs

```diff
@@ -353,12 +360,112 @@ impl RpcApi for Client {
         Ok(self.ledger.get_block_height()?.into())
     }
+
+    fn fund_raw_transaction<R: bitcoincore_rpc::RawTx>(
+        &self,
+        tx: R,
+        _options: Option<&json::FundRawTransactionOptions>,
+        _is_witness: Option<bool>,
+    ) -> bitcoincore_rpc::Result<json::FundRawTransactionResult> {
+        let mut transaction: Transaction = encode::deserialize_hex(&tx.raw_hex())?;
+        tracing::debug!("Decoded input transaction: {transaction:?}");
+
+        let mut hex: Vec<u8> = Vec::new();
+        let tx = encode_to_hex(&transaction);
+        tx.consensus_encode(&mut hex).unwrap();
+
+        let diff = match self.ledger.check_transaction_funds(&transaction) {
+            // If input amount is sufficient, no need to modify anything.
+            Ok(()) => {
+                return Ok(json::FundRawTransactionResult {
+                    hex,
+                    fee: Amount::from_sat(0),
+                    change_position: -1,
+                })
+            }
+            // Input funds are lower than the output funds, use the difference.
+            Err(LedgerError::InputFundsNotEnough(diff)) => diff,
+            // Other ledger errors.
+            Err(e) => return Err(e.into()),
+        };
+
+        tracing::debug!(
+            "Input funds are {diff} sats lower than the output sats, adding new input."
+        );
+
+        // Generate a new txout.
+        let address = self.get_new_address(None, None)?.assume_checked();
+        let txid = self.send_to_address(
+            &address,
+            Amount::from_sat(diff * diff),
+            None,
+            None,
+            None,
+            None,
+            None,
+            None,
+        )?;
+
+        let txin = self.ledger.create_txin(txid, 0);
+
+        transaction.input.insert(0, txin);
+        tracing::debug!("New transaction: {transaction:?}");
+
+        let tx = encode_to_hex(&transaction);
+        let mut hex: Vec<u8> = Vec::new();
+        tx.consensus_encode(&mut hex).unwrap();
+
+        Ok(json::FundRawTransactionResult {
+            hex,
+            fee: Amount::from_sat(0),
+            change_position: 0,
+        })
+    }
+
+    fn sign_raw_transaction_with_wallet<R: bitcoincore_rpc::RawTx>(
+        &self,
+        tx: R,
+        _utxos: Option<&[json::SignRawTransactionInput]>,
+        _sighash_type: Option<json::SigHashType>,
+    ) -> bitcoincore_rpc::Result<json::SignRawTransactionResult> {
+        let mut transaction: Transaction = encode::deserialize_hex(&tx.raw_hex())?;
+        tracing::debug!("Decoded input transaction: {transaction:?}");
+
+        let mut credentials = UserCredential::new();
+        ledger::Ledger::create_witness(&mut credentials);
+
+        let inputs: Vec<TxIn> = transaction
+            .input
+            .iter()
+            .map(|input| {
+                let mut input = input.to_owned();
+
+                if input.witness.is_empty() {
```

Comment:
We also need to check if the scriptPubkey matches with our constant address

Reply:
Fixed it with [f461deb](https://github.com/chainwayxyz/bitcoin-mock-rpc/pull/51/commits/f461deb6e3409d51b49277239db1845b5619f472).